ngraph/frontend/paddlepaddle/src/op/reshape2.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "reshape2.hpp"
+#include <ngraph/opsets/opset6.hpp>
+#include <paddlepaddle_frontend/utility.hpp>
+
+namespace ngraph
+{
+    namespace frontend
+    {
+        namespace pdpd
+        {
+            namespace op
+            {
+                NamedOutputs reshape2(const NodeContext& node)
+                {
+                    auto data = node.get_ng_input("X");
+                    if (!node.has_ng_input("Shape") && !node.has_ng_input("ShapeTensor"))
+                    {
+                        auto shape_attr = node.get_attribute<std::vector<int32_t>>("shape");
+                        auto shape_node = ngraph::opset6::Constant::create(
+                            ngraph::element::i32, {shape_attr.size()}, shape_attr);
+                        return node.default_single_output_mapping(
+                            {std::make_shared<ngraph::opset6::Reshape>(data, shape_node, true)},
+                            {"Out"});
+                    }
+                    else
+                    {
+                        std::string name = "Shape";
+                        if (node.has_ng_input("ShapeTensor"))
+                        {
+                            name = "ShapeTensor";
+                        }
+
+                        auto nodes = node.get_ng_inputs(name);
+                        ngraph::NodeVector node_vec;
+                        for (auto& input_node : nodes)
+                        {
+                            auto cast =
+                                std::make_shared<ngraph::opset6::Convert>(input_node, element::i64);
+                            node_vec.push_back(cast);
+                        }
+
+                        auto shape_node = std::make_shared<ngraph::opset6::Concat>(node_vec, 0);
+                        return node.default_single_output_mapping(
+                            {std::make_shared<ngraph::opset6::Reshape>(data, shape_node, true)},
+                            {"Out"});
```

Comment:
openvino Reshape treats zeros differently if special_zero attribute is set, is it intentional? Also I see some code in the doc linked in the PR description -- with that the translation seems incomplete. Are there any other sources to understand the semantics of this operation?

Reply:
@mangguo321 Please provide more info for explanation

Reply:
1.  For `sepcial_zero`, yes, it's needed. Paddle will copy the input dim when specifying zeros. Refer [here](https://github.com/PaddlePaddle/Paddle/blob/0dd6a44acb30cc708db22faf56f24c11cee96831/python/paddle/tensor/manipulation.py#L1584) and [here](https://github.com/PaddlePaddle/Paddle/blob/758dd7bb8f560e45bc49855e4776a9d64c95aea9/paddle/fluid/operators/reshape_op.cc#L247)
2.  For translation, the Paddle Reshape Ops has some hidden attributes that we cannot use. I check the Paddle Doc, the more accurate description should be [here](https://github.com/PaddlePaddle/Paddle/blob/758dd7bb8f560e45bc49855e4776a9d64c95aea9/paddle/fluid/operators/reshape_op.cc#L247)